In [1]:
pip install statsmodels


[notice] A new release of pip available: 22.3.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip



     ---------------------------------------- 9.9/9.9 MB 1.9 MB/s eta 0:00:00
     ---------------------------------------- 46.2/46.2 MB 4.8 MB/s eta 0:00:00
     -------------------------------------- 233.9/233.9 kB 4.8 MB/s eta 0:00:00


In [2]:
pip install scipy

Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip available: 22.3.1 -> 24.0
[notice] To update, run: python.exe -m pip install --upgrade pip


In [7]:
import numpy as np

## Task1
Suppose a child psychologist claims that the average time working mothers spend talking to their children 
is at least 11 minutes per day.You conduct a random sample of 1000 working mothers and find they spend an 
average of 11.5 minutes per day talking with their children.Assume prior research suggests the population standard deviation is 2.3 minutes.
Conduct a test with a level of significance of alpha = 0.05.


Null Hypothesis Ho = μ ≤ 11 minutes


Alternative Hypothesis Ha = μ > 11 minutes


x̄ = 11.5 minutes


n = 1000


σ = 2.3 minutes


α = 0.05


μ = 11 minutes


Since we have a sample of 1000, which is greater than 30, and we have a sample mean, standard deviation, and 
want to compare the sample mean to a population value ,so we can use Z-test.

Z=(x̄ - μ)/(σ/√n)

In [3]:
import math
Z = (11.5-11)/(2.3/math.sqrt(1000))
Z

6.874516652539955

crirical value for a z-score corresponding to an alpha level of 0.05 from z-distribution table = 1.645.
It represents the value beyond which you would reject the null hypothesis.


## INFERENCE


### z-score( 6.8) >  criticalvalue( 1.64), so reject the null hypothesis .That means the average time working mothers spend talking to their children is indeed more than 11 minutes per day.

## Task 2

A coffee shop claims that their average wait time for customers is less than 5 minutes. To test this claim, a sample of 40 customers is taken, and their wait times are recorded. The sample mean wait time is found to be 4.6 minutes with a standard deviation of 0.8 minutes. Perform a hypothesis test at a significance level of 0.05 and determine whether there is enough evidence to support the coffee shop's claim.



### NULL HYPOTHESIS
Ho = μ ≥ 5 minutes

### ALTERNATIVE HYPOTHESIS
Ha = μ < 5 minute


x̄ = 4.6 minutes


n = 40


σ(sample) = 0.8 minutes


α = 0.05


Degree of freedom(n-1)= 39


μ = 5 minutesutess


When the population standard deviation is unknown, the t-test is suitable, especially for smaller sample sizes, typically less than 30. However, even for larger sample sizes, typically greater than 30, the t-test remains applicable despite the unknown population standard deviation, owing to the Central Limit Theorem. This theorem states that as the sample size increases, the distribution of sample means tends towards a normal distribution.


t = (x̄ - μ) / (σ / √n)

In [4]:
t = (4.6-5)/(0.8/math.sqrt(40))
t

-3.162277660168382

The function t.ppf(alpha, df) calculates a critical value from the t-distribution based on the provided significance level (alpha) and degrees of freedom (df).The term "percent point function" (ppf) means it gives you a value from the distribution such that if you randomly select a value from the distribution, there's a certain probability it will be less than or equal to that value.

In [5]:
#calculating the critical value with df 39 and alpha 0.05
from scipy.stats import t
critical_value = t.ppf(0.05, 39)
print("Critical value:", critical_value)

Critical value: -1.684875121711225


### Conclusion

Calculated t-value, which is -3.16, falls below the critical t-value of -1.685. Therefore, we reject the null hypothesis. This implies there is sufficient evidence to support the assertion that the average wait time for customers is less than 5 minutes